groth16/src/prover.rs

```diff
@@ -242,58 +155,63 @@ where
     let (full_input_assignment, h, _) = R1CStoQAP::witness_map::<E>(&prover)?;
     end_timer!(witness_map_time);
 
-    let input_assignment = full_input_assignment[1..prover.num_inputs]
-        .into_iter()
+    let num_inputs = prover.input_assignment.len();
+
+    let input_assignment = cfg_into_iter!(full_input_assignment[1..num_inputs])
```

Comment:
Because inputs are usually small, and the into_repr computation pretty cheap, this wouldn’t really help, and might even harm things because of the overhead due to synchronization.

Reply:
Good point, fixed https://github.com/scipr-lab/zexe/pull/171/commits/2d9fe6ab08593cc44f578686b6787d9b5acd9343#diff-4a7e8a6b598180387474d3d2ad21ab90R162-R163